gui/pulsed/pulsed_maingui.py

```diff
@@ -3103,4 +3172,8 @@ def update_laser_data(self):
         self.lasertrace_image.setData(x=x_data, y=y_data)
         return
 
+    @QtCore.Slot()
+    def run_pg_benchmark(self):
 
+        self.pulsedmasterlogic().sequencegeneratorlogic().sigRunPgBenchmark.emit()
+        self.pulsedmasterlogic().sigGeneratorSettingsChanged.emit({})
```

Comment:
Why is this signal emitted here?

Reply:
Thanks for the in-depth review. I will have a look at the issues raised.
I agree, the benchmark tool should probably live outside the sequence generator logic, but I didn't find an obvious well-suited place. Any suggestion?

Reply:
You could simply move the benchmarking code to a separate file, e.g. `qudi/logic/pulsed/_pulser_benchmark.py`. It could contain a class or a function to run the benchmark that needs to be handed over the `SequenceGeneratorLogic` instance.
Something along these lines:

``` python
from ._pulser_benchmark import MyBenchmarkTool

...

class SequenceGeneratorLogic(LogicBase):
...
    # somewhere in SequenceGeneratorLogic
    def run_benchmark(self):
        tool = MyBenchmarkTool(the_logic=self)
        result = tool.run()
        ...
```